File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/RiskAssessment.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class RiskAssessment : DomainResource, IResource
{
    public Base.ResourceReference? Parent { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public RiskAssessment_Prediction[]? Prediction { get; set; }
    public Base.CodeableConcept? Method { get; set; }
    public Base.ResourceReference[]? Basis { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public string? Mitigation { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.Period? OccurrencePeriod { get; set; }
    public required string Status { get; set; }
    public Base.ResourceReference? Condition { get; set; }
    public Base.CodeableConcept? Code { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference? BasedOn { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public required Base.ResourceReference Subject { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrencePeriod is not null)
            {
                return OccurrencePeriod;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Period))
            {
                OccurrencePeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference? Performer { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class RiskAssessment_Prediction : BackboneElement
    {
        public string? RelativeRisk { get; set; }
        public object? When    
        {
            get
            {
                if (WhenRange is not null)
                {
                    return WhenRange;
                }
        
                if (WhenPeriod is not null)
                {
                    return WhenPeriod;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.Range))
                {
                    WhenRange = (Base.Range)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Period))
                {
                    WhenPeriod = (Base.Period)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.Range? WhenRange { get; set; }
        public Base.CodeableConcept? Outcome { get; set; }
        public object? Probability    
        {
            get
            {
                if (ProbabilityRange is not null)
                {
                    return ProbabilityRange;
                }
        
                if (ProbabilityDecimal is not null)
                {
                    return ProbabilityDecimal;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(Base.Range))
                {
                    ProbabilityRange = (Base.Range)value;return;
                }
        
                if (value?.GetType() == typeof(string))
                {
                    ProbabilityDecimal = (string)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public Base.Period? WhenPeriod { get; set; }
        public string? Rationale { get; set; }
        public Base.Range? ProbabilityRange { get; set; }
        public Base.CodeableConcept? QualitativeRisk { get; set; }
        public string? ProbabilityDecimal { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/RiskAssessment.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Range
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class RiskAssessmentPrediction(BackboneElement):
    relative_risk: Optional[float] = None
    when_range: Optional[Range] = None
    outcome: Optional[CodeableConcept] = None
    when_period: Optional[Period] = None
    rationale: Optional[str] = None
    probability_range: Optional[Range] = None
    qualitative_risk: Optional[CodeableConcept] = None
    probability_decimal: Optional[float] = None

@dataclass(kw_only=True)
class RiskAssessment(DomainResource):
    parent: Optional[Reference] = None
    encounter: Optional[Reference] = None
    prediction: Optional[List[RiskAssessmentPrediction]] = field(default_factory=list)
    method: Optional[CodeableConcept] = None
    basis: Optional[List[Reference]] = field(default_factory=list)
    reason_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    mitigation: Optional[str] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    occurrence_period: Optional[Period] = None
    status: str
    condition: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    based_on: Optional[Reference] = None
    occurrence_date_time: Optional[str] = None
    subject: Reference
    performer: Optional[Reference] = None
    reason_reference: Optional[List[Reference]] = field(default_factory=list)